app/error_messages.py

```diff
@@ -4,6 +4,9 @@
 ANSWER_REFERENCE_CANNOT_BE_USED_ON_MIN = (
     "The referenced answer cannot be used to set the minimum of answer"
 )
+ANSWER_OPTION_LABEL_VALUE_TYPE_INVALID = (
+    "The referenced answer_id type is not of type ['Radio','Checkbox','Dropdown']"
```

Comment:
Given this is specific, it would be good to be a bit more explicit, as it didn't have enough context.  We can also reuse this in rules validator when we come to do that.
```suggestion
    "The referenced answer for option label from value placeholder/operator is not of type ['Radio','Checkbox','Dropdown']"
```